src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -403,6 +405,124 @@ public async Task CreateAsync(ContentItem contentItem, VersionOptions options)
+                    .ListAsync();
+
+                foreach (var version in versions)
+                {
+                    await LoadAsync(version);
+                }
+
+                // Query for versions that may be evicted.
+                var publishedLatestVersions = versions.Where(x => x.Latest && x.Published);
+                var reducedContentItems = batchedContentItems.Except(publishedLatestVersions);
+                var possibleEvictionIds = reducedContentItems.Select(x => x.ContentItemId);
+
+                var possibleEvictionVersions = await _session.Query<ContentItem, ContentItemIndex>()
+                    .Where(x => x.ContentItemId.IsIn(possibleEvictionIds) && (x.Latest || x.Published))
+                    .ListAsync();
+
+                foreach (var version in possibleEvictionVersions)
+                {
+                    await LoadAsync(version);
+                }
+
+                foreach (var importingItem in batchedContentItems)
+                {
+                    ContentItem originalVersion = null;
+                    if (!String.IsNullOrEmpty(importingItem.ContentItemVersionId))
+                    {
+                        originalVersion = versions.FirstOrDefault(x => String.Equals(x.ContentItemVersionId, importingItem.ContentItemVersionId, StringComparison.OrdinalIgnoreCase));
+                    }
+
+                    if (originalVersion == null)
+                    {
+                        // The version does not exist in the current database.
+                        var result = await CreateContentItemVersionAsync(importingItem, possibleEvictionVersions.Where(x => String.Equals(x.ContentItemId, importingItem.ContentItemId, StringComparison.OrdinalIgnoreCase)));
+                        if (!result.Succeeded)
+                        {
+                            if (_logger.IsEnabled(LogLevel.Error))
+                            {
+                                _logger.LogError("Error importing content item version id '{ContentItemVersionId}' : '{Errors}'", importingItem?.ContentItemVersionId, string.Join(',', result.Errors));
+                            }
+
+                            throw new Exception(string.Join(',', result.Errors));
+                        }
+                    }
+                    else
+                    {
+                        // The version exists in the database.
+                        // We compare the two versions and skip importing it if they are the same.
+                        // We do this to prevent unnecessary sql updates, and because UpdateContentItemVersionAsync
+                        // will remove drafts of updated items to prevent orphans.
+                        // So it is important to only import changed items.
+                        var jImporting = JObject.FromObject(importingItem);
+                        var jOriginal = JObject.FromObject(originalVersion);
+
```

Comment:
This comparison is important, because the deployment step will export all published and latest content, even if it has not changed (how would it know that detail...)

So this will no-op if the `ContentItem` has not changed.